public/app/plugins/panel/nodeGraph/layout.worker.utils.js

```diff
@@ -0,0 +1,173 @@
+// This file exists so we can import the layout function for mocks in jest without
+// errors caused by addEventListener.
```

Comment:
nit: Isn't it possible to just mock the addEventListener in tests?

Reply:
Also it wasn't a problem before right so it is because of the changes in webpack or something?

Reply:
Thanks for taking the time to look at this @aocenas. The issue I was working around here seemed to be due to [this mocks import](https://github.com/grafana/grafana/blob/main/public/test/mocks/workers.ts#L4). For some reason Jest started failing to run due to the addEventListener so moved the layout logic into a separate file and imported that. I can remove it if you're not keen and we can figure out a different way forward. 👍

Reply:
I believe it's related to the changes here that mean monaco and kusto aren't mocked in tests anymore. I'm pretty sure their service workers were somehow causing this to fail. Let me try to reproduce it again and I'll add the error here so we can better understand the issue. (It's been months since I did this work.)

Reply:
Yeah I mean not a big deal, just wondering what the root cause is and whether this will be needed for all service worker code or not. Also will be moving this to a separate package so want to know if it will need to be done the same way there (or maybe not because of different webpack build or something)

Reply:
>Yeah I mean not a big deal, just wondering what the root cause is and whether this will be needed for all service worker code or not. Also will be moving this to a separate package so want to know if it will need to be done the same way there (or maybe not because of different webpack build or something)

I guess somewhere we need to guard against undefined being passed to layout or within the layout function we return early?